### PR TITLE
feat: use store-cli to upload artifacts instead of curl

### PR DESCRIPTION
**Pull request by s-yoshika**
In the [store-cli PR](https://github.com/screwdriver-cd/store-cli/pull/31) we can use store-cli to upload artifacts.

This PR switch to store-cli from curl for uploading artifacts.
This PR is blocked until store-cli PR is merged and new launcher is built.

I confirmed this works well in local darwin environment.
```
$ export SD_ZIP_ARTIFACTS=false
$ bash commands.txt
2018/12/07 16:10:31 Upload to http://172.16.164.207:9002/v1/builds/6/ARTIFACTS/coverage%2Fcoverage.json successful.
2018/12/07 16:10:31 Upload to http://172.16.164.207:9002/v1/builds/6/ARTIFACTS/coverage%2Flcov-report%2Fartifact-bookend%2Findex.html successful.
2018/12/07 16:10:31 Upload to http://172.16.164.207:9002/v1/builds/6/ARTIFACTS/coverage%2Flcov-report%2Fartifact-bookend%2Findex.js.html successful.
2018/12/07 16:10:31 Upload to http://172.16.164.207:9002/v1/builds/6/ARTIFACTS/coverage%2Flcov-report%2Fbase.css successful.
...
...
$ export SD_ZIP_ARTIFACTS=true
$ bash commands.txt
updating: html/index.html (stored 0%)
updating: html/あいうえお.txt (stored 0%)
updating: manifest.txt (deflated 59%)
updating: README.md (deflated 62%)
  adding: coverage/coverage.json (deflated 73%)
  adding: coverage/lcov-report/artifact-bookend/index.html (deflated 72%)
  adding: coverage/lcov-report/artifact-bookend/index.js.html (deflated 74%)
  adding: coverage/lcov-report/base.css (deflated 68%)
  adding: coverage/lcov-report/index.html (deflated 72%)
  adding: coverage/lcov-report/prettify.css (deflated 55%)
  adding: coverage/lcov-report/prettify.js (deflated 60%)
  adding: coverage/lcov-report/sort-arrow-sprite.png (stored 0%)
  adding: coverage/lcov-report/sorter.js (deflated 73%)
  adding: coverage/lcov.info (deflated 44%)
  adding: test/xunit.xml (deflated 50%)
2018/12/07 16:11:30 Upload to http://172.16.164.207:9002/v1/builds/6/ARTIFACTS/SD_ARTIFACT.zip successful.
```

### CODE REVIEW

**Review by sakka2 — APPROVED** [review]
LGTM

**Review by yuichi10 — APPROVED** [review]
LGTM

File: commands.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
...
+if [ "$SD_ZIP_ARTIFACTS" = "true" ]; then ZIPFILE=SD_ARTIFACT.zip && find . -type f -exec $SD_ZIP_CMD_WRAPPER zip $ZIPFILE {} + && store-cli set $ZIPFILE --type=artifact; else find . -type f -exec store-cli set $1 --type=artifact -- {} \;; fi;
```
Comment (jithine) [inline commands.txt:4]: does store cli version take care of
1.  ~retry logic ?~
2. Using http1.1 ? We defaulted to 1.1 when we were observing upload errors wrt default http2.0 .

Comment (d2lam) [reply commands.txt:4]: @jithin1987  `store-cli` does have retry logic: https://github.com/screwdriver-cd/store-cli/blob/master/sdstore/sdstore.go#L100-L112

Comment (tk3fftk) [reply commands.txt:4]: @jithin1987 
`store-cli` is using http 1.1 via Go-http-client. Below log is from our development environment.
```
nginx-ingress-controller-vqjz2 nginx-ingress-lb  xxx.xxx.xxx.xxx - [ xxx.xxx.xxx.xxx] - - [10/Dec/2018:07:38:23 +0000] "PUT /v1/builds/9954/ARTIFACTS/html%2Findex.html HTTP/1.1" 202 0 "-" "Go-http-client/1.1" 772 0.088 [sdstore-80] xxx.xxx.xxx.xxx:80 0 0.088 202
```

Reference: https://golang.org/pkg/net/http/#Transport
> Transport uses HTTP/1.1 for HTTP URLs and either HTTP/1.1 or HTTP/2 for HTTPS URLs, depending on whether the server supports HTTP/2, and how the Transport is configured.

Comment (catto) [reply commands.txt:4]: I also saw an upload error when HTTP/2 was enabled.
> curl: (92) HTTP/2 stream 1 was not closed cleanly: PROTOCOL_ERROR (err 1)

I confirmed  that there is no error with `store-cli` in our environment so far.

**Review by catto — APPROVED** [review]

**Review by jithine — APPROVED** [review]

**Comment by jithine** [comment]
@d2lam let's keep an eye on this when we pull this internally to make sure that our ingress is not using http2. We were forcing it previously.

**Pull request merged**